app/models/condition.rb

```diff
@@ -1,3 +1,112 @@
 class Condition < ApplicationRecord
   validates :date, presence: true
+
```

Comment:
Question re: acceptable balance of AR and Ruby: **If ActiveRecord is handling all the groupings and counts in the conditions dash, is it okay to use Ruby to find the min/max/average of those counts?**  

The `max_trips_by`,`min_trips_by` and `avg_trips_by` methods in this model iterate over the `<attribute>_ranges` hashes, sending the values to one of the four ActiveRecord query methods at the top (e.g. `trips_by_temperature` or `trips_by_precipitation`). The query methods return a hash where keys are dates and the values are trip count. 

This AR hash is then passed to the `return_min`, `return_max`, and `return_avg` methods, which return the highest/lowest/avg value in the hash.

Now, the `return_min` and `return_max` methods _could_ be replaced by building out longer queries.  Such as using the following to get max rides by day by visibility:

```
Condition.joins("JOIN trips ON conditions.date = trips.start_date")
  .select("conditions.date, count(trips.id) as trip_count")
  .where(mean_visibility_miles: [10..20])
  .group("conditions.date")
  .order("trip_count DESC")
  .first
  .trip_count
```
However, I like that in it's current construction, we are always dealing with the same ActiveRecord hash and leaning on some explicit Ruby methods to return the count we need. The query above is arguably more convoluted, all for the purpose of returning an integer from a hash. What are your thoughts on how this has been put together?